opm/output/eclipse/RestartIO.cpp

```diff
@@ -654,6 +680,64 @@ namespace {
                              std::forward<OutputVectorInt>(writeVectorI));
     }
 
+    void writeFluidInPlace(const RestartValue&           value,
+                           const EclipseState&           es,
+                           const bool                    writeDouble,
+                           EclIO::OutputStream::Restart& rstFile)
+    {
+        const auto vectors = fluidInPlaceVectorNames(value);
+
+        if (vectors.empty()) {
+            return;
+        }
+
+        {
+            auto regSets = es.fieldProps().fip_regions();
+            std::sort(regSets.begin(), regSets.end());
+            rstFile.write("FIPFAMNA", regSets);
+        }
+
+        auto writeVector =
+            [writeDouble, &rstFile](const std::string&         arrayName,
+                                    const std::vector<double>& fipArray)
+        {
+            if (writeDouble) {
+                rstFile.write(arrayName, fipArray);
+            }
+            else {
+                rstFile.write(arrayName, std::vector<float> {
+                    fipArray.begin(), fipArray.end()
+                });
+            }
+        };
+
+        auto anyRSFip = false;
+        for (const auto& vector : vectors) {
+            writeVector(vector, value.solution.at(vector).data<double>());
+
+            if ((vector.front() == 'R') || (vector.front() == 'S')) {
+                // The vector name is RFIP* or SFIP*.  These refer to
+                // reservoir and surface condition volumes, respectively,
+                // meaning the simulator provides in-place arrays that have
+                // been properly tagged.
+                anyRSFip = true;
+            }
+        }
+
+        if (anyRSFip) {
+            // The simulator provides properly tagged in-place arrays.  No
+            // further action needed.
+            return;
+        }
+
+        // If we get here, all fluid-in-place vectors have the name FIP* and
+        // represent surface condition volumes.  Output the same vectors
+        // using the correaponding SFIP name as well.
```

Comment:
corresponding

Reply:
> corresponding

Thanks.  I've pushed an update to fix that.